### PR TITLE
[Bug fix] restore test for splitting video

### PR DESCRIPTION
**Pull request by wonjuleee**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/integration/cli/test_utils.py [hunk 1]
```diff
@@ -40,7 +39,7 @@ def test_can_split_video(self):
             "--start-frame",
             "2",
             "--end-frame",
-            "8",
+            "7",
```
Comment (vinnamkim) [inline tests/integration/cli/test_utils.py]: I guess there are some non-deterministic processes during video encoding and decoding. Can we make video frame grabbing deterministic using `mock.patch`? For example,
```suggestion
with mock.patch("datumaro.components.media.VideoFrame.data") as mock_video_frame_data:
     mock_video_frame_data.side_effect = [np.full(value=idx, size=(3,32,32), dtype=np.uint8) for frame_idx in range(10)]
     run(
         ...
     )
```

Comment (wonjuleee) [reply tests/integration/cli/test_utils.py]: Thanks Vinnam, I have updated the test as following your guide. Please review again when you are available.

File: src/datumaro/components/media.py [hunk 2]
```diff
@@ -355,7 +355,11 @@ def save(
         ext: Optional[str] = None,
         crypter: Crypter = NULL_CRYPTER,
     ):
-        data = self.data
+        if callable(self.data):
+            data = np.array(self.data())
+        else:
+            data = self.data
```
Comment (vinnamkim) [inline src/datumaro/components/media.py]: Does `ImageFromData` already have this logic as follow (inherited by `FromDataMixin`)?

https://github.com/openvinotoolkit/datumaro/blob/ea19ad2f3801a10b033e30e70c74215333eeb34a/src/datumaro/components/media.py#L179-L183

Comment (wonjuleee) [reply src/datumaro/components/media.py]: Thank you for your help. I updated the mock video frame generation with propertymock.

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Pull request merged**